File: topic_5/Python/4-maioresQue.py
# Recebe um numero n e uma lista e retorna uma sublista de numeros que
# são maiores que n da lista

def maioresQue(n, xs):
    lista = []
    for x in xs:
        if x > n:
            lista.append(x)
    return lista

print(maioresQue(10, []) == []) #True
print(maioresQue(10, [11,9,12]) == [11,12]) #True
print(maioresQue(10, [4,6,30,3,15,3,10,7]) == [30,15]) #True

File: topic_5/Haskell/4-maioresQue.hs
-- Recebe um numero n e uma lista e retorna uma sublista de numeros que
-- são maiores que n da lista

maioresQue n [] = []
maioresQue n xs = filter (> n) xs

main = do
    a <- readLn :: IO Int
    b <- readLn :: IO [Int]
    print $ maioresQue a b